posix/posix.c

```diff
@@ -477,6 +477,78 @@ static int posix_create(const char *filename, int type, mode_t mode, oid_t dev,
 	return err;
 }
 
+int posix_statvfs(const char *path, int fildes, struct statvfs *stat)
+{
+	oid_t oid, dev;
+	oid_t *oidp, *devp;
+	open_file_t *f;
+	msg_t msg;
+	int err = EOK;
+
+	if (((path == NULL) && (fildes < 0)) ||
+			((path != NULL) && (fildes != -1))) {
+		return -EINVAL;
+	}
+
+	if (path == NULL) {
+		err = posix_getOpenFile(fildes, &f);
+		if (err < 0) {
+			return err;
+		}
+		oidp = &f->oid;
+		devp = NULL;
+	}
+	else {
+		if (proc_lookup(path, &oid, &dev) < 0) {
+			return -ENOENT;
+		}
+		oidp = &oid;
+		devp = &dev;
+	}
+
+	/* Detect mountpoint */
+	if ((devp != NULL) && (oidp->port != devp->port)) {
+		hal_memset(&msg, 0, sizeof(msg));
+		msg.type = mtGetAttr;
+		hal_memcpy(&msg.oid, oidp, sizeof(*oidp));
+		msg.i.attr.type = atMode;
+
+		if ((proc_send(oidp->port, &msg) < 0) || (msg.o.err < 0)) {
+			(void)posix_fileDeref(f);
```

Comment:
What if `f` is used uninitialized? If `posix_getOpenFile()` is performed with `&f`, then `devp == NULL`, so this branch isn't even taken when `f` is used.

Reply:
good catch